src/app/spreed-webrtc-server/channelling_api.go

```diff
@@ -60,169 +60,113 @@ func NewChannellingAPI(config *Config, roomStatus RoomStatusManager, sessionEnco
 	}
 }
 
-func (api *channellingAPI) OnConnect(client Client, session *Session) {
+func (api *channellingAPI) OnConnect(client Client, session *Session) (interface{}, error) {
 	api.Unicaster.OnConnect(client, session)
-	api.SendSelf(client, session)
+	return api.MakeSelf(session)
 }
 
 func (api *channellingAPI) OnDisconnect(client Client, session *Session) {
 	api.Unicaster.OnDisconnect(client, session)
 }
 
-func (api *channellingAPI) OnIncoming(c ResponseSender, session *Session, msg *DataIncoming) {
+func (api *channellingAPI) OnIncoming(sender Sender, session *Session, msg *DataIncoming) (interface{}, error) {
 	switch msg.Type {
 	case "Self":
-		api.SendSelf(c, session)
+		return api.MakeSelf(session)
```

Comment:
+1 for HandleSelf change. Rest looks good, and i merge as soon as HandleSelf is there.